catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -0,0 +1,196 @@
+            @NotNull CatalogMigratableConfig config) {
+        super(description, true);
+
+        notNull(provider, "CatalogProvider cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+
+        this.provider = provider;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+
+        this.taskManager = createTaskManager(config, createExecutorService(config));
+    }
+
+    /**
+     * Exports all the metacards currently stored in the catalog provider.
+     * <p>
+     * {@inheritDoc}
+     */
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        config.setExportPath(exportPath.resolve(CATALOG_EXPORT_DIRECTORY));
+        fileWriter.createExportDirectory(config.getExportPath());
+
+        Collection<MigrationWarning> warnings = new ArrayList<>();
+        Map<String, Serializable> props = new HashMap<>();
+        Filter dumpFilter = filterBuilder.attribute(Metacard.ANY_TEXT)
+                .is()
+                .like()
+                .text("*");
+
+        useNativeQueryMode(props);
+
+        QueryImpl exportQuery = createQuery(dumpFilter);
+        exportQuery.setPageSize(config.getExportQueryPageSize());
+        exportQuery.setRequestsTotalResultsCount(false);
+
+        QueryRequest exportQueryRequest = new QueryRequestImpl(exportQuery, props);
+
+        try {
+            executeQueryLoop(exportQuery, exportQueryRequest);
+        } catch (UnsupportedQueryException e) {
+            LOGGER.error("Query {} was invalid due to: {}", exportGroupCount, e.getMessage(), e);
+            throw new ExportMigrationException("Catalog could not export metacards");
+        } catch (RuntimeException e) {
+            LOGGER.error("Internal error occurred when exporting catalog: {}", e.getMessage(), e);
+            throw new ExportMigrationException("Catalog could not export metacards");
+        } finally {
+            try {
+                taskManager.exportFinish();
+            } catch (RuntimeException e) {
+                throw new ExportMigrationException(e);
+            }
+        }
+
+        return new MigrationMetadata(warnings);
+    }
+
+    // Factory method for creating QueryImpl objects so they can be mocked later.
+    QueryImpl createQuery(Filter dumpFilter) {
+        return new QueryImpl(dumpFilter);
+    }
+
+    // Factory method for creating MigrationTaskManager objects so they can be mocked later.
+    MigrationTaskManager createTaskManager(CatalogMigratableConfig config,
+            ExecutorService executorService) {
+        return new MigrationTaskManager(config, fileWriter, executorService);
+    }
+
+    private ExecutorService createExecutorService(CatalogMigratableConfig config) {
+        return new ThreadPoolExecutor(config.getExportThreadCount(),
```

Comment:
You're going to keep a max count pool of idle threads around all the time with this. There's no good reason to set the core size so high.